app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtTextActions.java

```diff
@@ -322,4 +330,71 @@ public boolean onLongClick(View v) {
             return new ArrayList<>();
         }
     }
+
+    /**
+     * Insert or update a date
+     *
+     * This routine checks the word under the cursor and sees if it matches
+     * the todo.txt date format `tag:YYYY-mm-dd` or plain `YYYY-mm-dd`.
+     * If a date is found, it is parsed and displayed to the user to be updated.
+     *
+     * If no date is found, the user selected date will be inserted at the cursor.
+     * If provided, a key will be inserted before the date.
+     * (key = 'due => date inserted = `due:YYYY-mm-dd`)
+     *
+     * @param key An optional prefix key. optional (use null for no key)
+     */
+    protected void updateOrInsertTodoDate(final String key) {
```

Comment:
Suggestion for simplification:
* Use `String oldDue = SttCommander::parseDueDate()` function on task of current line
* String currentLine = ....; currentLine = currentLine.replace(oldDue, newDue);

Reply:
The function searches for a date _under the cursor_.

My use cases are
1. Change creation date (rare)
1. Change due date
2. Change threshold date

With the current approach, I can tap on a date, tap the 'date' action button and quickly change the date.

I think what you are suggesting is that we detect and insert or replace the due date in the current line. Am I correct?